.github/workflows/linters-tests-image.yaml

```
@@ -0,0 +1,114 @@
+name: linters-tests-image
+
+on:
+  pull_request:
+    branches: [ '*' ]
+  push:
+    branches:
+      - "main"
+    tags:
+      - 'v*'
+
+env:
+  REGISTRY: ghcr.io
+  REPOSITORY: registry-operator
+  IMAGE: registry-operator
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: wagoid/commitlint-github-action@v5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: ./go.mod
+          cache: false
+      - uses: golangci/golangci-lint-action@v3
```

Consider using latest version here

```suggestion
      - uses: golangci/golangci-lint-action@v4
```